### PR TITLE
fix: Provide signing arguments as command line parameters

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Related to https://github.com/appium/appium/issues/19190

### CODE REVIEW

**Review by KazuCocoa — APPROVED** [review]

**Comment by mykola-mokhnach** [comment]
@KazuCocoa Could you please verify the patch with real devices before I merge it?

File: lib/xcodebuild.js [hunk 1]
```diff
@@ -238,9 +238,18 @@ class XcodeBuild {
...
-      args.push('-xcconfig', this.xcodeConfigFile);
+    if (this.realDevice) {
+      if (this.xcodeConfigFile) {
+        this.log.debug(`Using Xcode configuration file: '${this.xcodeConfigFile}'`);
+        args.push('-xcconfig', this.xcodeConfigFile);
+      }
+      if (this.xcodeOrgId && this.xcodeSigningId) {
+        args.push(
+          `DEVELOPMENT_TEAM=${this.xcodeOrgId}`,
+          `CODE_SIGN_IDENTITY=${this.xcodeSigningId}`,
```
Comment (KazuCocoa) [inline lib/xcodebuild.js:249]: oh, one thing, probably needs `"`? since it could include spaces

**Comment by KazuCocoa** [comment]
yes, sure. The arguments themselves seem good, but let me run on my local

File: lib/xcodebuild.js (hunk 1 above)
Comment (mykola-mokhnach) [reply lib/xcodebuild.js:249]: exec should escape it automatically as these are separate array items

File: lib/xcodebuild.js [hunk 2]
```diff
@@ -238,9 +238,18 @@ class XcodeBuild {
...
+          'CODE_SIGN_STYLE=Manual'
```
Comment (KazuCocoa) [inline lib/xcodebuild.js]:
```suggestion
```

it seems like this should not exist. Build failed as 70 with `CODE_SIGN_STYLE=Manual`, but succeeded without that. Maybe CODE_SIGN_STYLE=Manual requires more something.


Btw, when the build failure occurred, the below check continued even response back the error message to the client. Maybe we need to tweak that later

```
[debug] [XCUITestDriver@6bc0] Proxying [GET /status] to [GET http://127.0.0.1:8100/status] with no body
[XCUITestDriver@6bc0] connect ECONNREFUSED 127.0.0.1:8100
[debug] [XCUITestDriver@6bc0] Matched '/status' to command name 'getStatus'
[debug] [XCUITestDriver@6bc0] Proxying [GET /status] to [GET http://127.0.0.1:8100/status] with no body
[XCUITestDriver@6bc0] connect ECONNREFUSED 127.0.0.1:8100
[debug] [XCUITestDriver@6bc0] Matched '/status' to command name 'getStatus'
[debug] [XCUITestDriver@6bc0] Proxying [GET /status] to [GET http://127.0.0.1:8100/status] with no body
[XCUITestDriver@6bc0] connect ECONNREFUSED 127.0.0.1:8100
[debug] [XCUITestDriver@6bc0] Matched '/status' to command name 'getStatus'
[debug] [XCUITestDriver@6bc0] Proxying [GET /status] to [GET http://127.0.0.1:8100/status] with no body
```

Comment (KazuCocoa) [reply lib/xcodebuild.js]: ^ tested with xcode 15 and 14.3.1

Comment (mykola-mokhnach) [reply lib/xcodebuild.js]: sure, thanks for checking

**Review by KazuCocoa — APPROVED** [review]

**Pull request merged**